### PR TITLE
Ajouter l'ADPA 93

### PR DESCRIPTION
**Pull request by fpagnoux**
Connected to #147

### CODE REVIEW

**Comment by MattiSG** [comment]
Remove reverts (and upstream merge if you rebased), squash commenting with the commented code (atomic commit = implementation + doc (+ possibly tests if they were not here already)).

**Comment by MattiSG** [comment]
On the image: apply the background blue on the white panels, to have a squared image.

File: app/js/controllers/foyer/individuForm.js [hunk 1]
```diff
@@ -122,6 +123,19 @@ angular.module('ddsApp').controller('FoyerIndividuFormCtrl', function($scope, in
         }
     };
 
+    $scope.capturePerteAutonomie = false;
+
+    /* We need to delay the apparition of the 'perteAutonomie' question,
+    so that the user has time to finish typing the date before it appears.
+    Otherwise, if the user types '1980', the question will appear after '19' is typed.
```
Comment (MattiSG) [inline app/js/controllers/foyer/individuForm.js]: > It was indeed important to comment, thanks for having done it. “we need to delay” is already readable in the code though, so a simple `// avoid displaying question when user born in 1980 is typing 19 as birth year` right next to the `, 400));` (or the `_.debounce`) would also make the fit  :)

File: app/js/controllers/foyer/individuForm.js [hunk 2]
```diff
@@ -122,6 +123,19 @@ angular.module('ddsApp').controller('FoyerIndividuFormCtrl', function($scope, in
         }
     };
 
+    $scope.capturePerteAutonomie = false;
+
+    /* We need to delay the apparition of the 'perteAutonomie' question,
```
Comment (MattiSG) [inline app/js/controllers/foyer/individuForm.js]: I'm afraid “apparition” is a _faux-ami_, “delay displaying” would probably be more appropriate  ;)

File: app/js/controllers/resultat.js [hunk 3]
```diff
@@ -39,8 +39,12 @@ angular.module('ddsApp').controller('ResultatCtrl', function($scope, $rootScope,
         });
     };
 
-    $scope.round = function(montant) {
-        return Math.round(montant / 10) * 10;
+    $scope.round = function(droit) {
+        if (! droit.unite) {
```
Comment (MattiSG) [inline app/js/controllers/resultat.js]: See sgmap/ludwig-ui#51 (why not `unit`?).

File: app/views/partials/foyer/individu-form.html [hunk 4]
```diff
@@ -266,6 +266,22 @@
     </div>
   </div>
 
+  <div class="form-group from-group-lg" ng-if="capturePerteAutonomie">
+    <label class="control-label col-sm-4" for="perteAutonomie">
+      Avez vous des difficultés pour accomplir les gestes de la vie quotidienne ?
```
Comment (MattiSG) [inline app/views/partials/foyer/individu-form.html]: “Avez**-**vous”

File: test/spec/controllers/resultat.js [hunk 5]
```diff
@@ -19,13 +19,13 @@ describe('Controller: ResultatCtrl', function() {
 
     it('round() should round to the nearest ten', function() {
```
Comment (MattiSG) [inline test/spec/controllers/resultat.js]: Doesn't fit anymore.

Split the test with `describe('round', describe('euros', it('should give nearest ten')), describe('percentage', it('should give nearest unit')))`.

**Comment by fpagnoux** [comment]
Reverts removed, but rebasing brings a lot of conflict, hence the upstream merge.

File: test/spec/controllers/resultat.js (hunk 5 above)
Comment (fpagnoux) [inline test/spec/controllers/resultat.js]: Done !

File: test/spec/controllers/resultat.js [hunk 6]
```diff
@@ -17,15 +17,30 @@ describe('Controller: ResultatCtrl', function() {
         });
     });
 
-    it('round() should round to the nearest ten', function() {
-        // given
-        var values = [100, 57, 54, 34.5];
-        controller('ResultatCtrl', { $scope: scope });
+    describe('round', function() {
+        describe('euros', function() {
+            it('should give nearet ten', function() {
```
Comment (MattiSG) [inline test/spec/controllers/resultat.js]: > nearest

File: test/spec/controllers/resultat.js [hunk 7]
```diff
@@ -17,15 +17,30 @@ describe('Controller: ResultatCtrl', function() {
...
+                var results = _.map(values, scope.round);
 
-        // then
-        expect(results).toEqual([100, 60, 50, 30]);
+                // then
+                expect(results).toEqual([100, 60, 50]);
+            });
+        });
+        describe('percentage', function() {
+            it('should give nearet unit', function() {
```
Comment (MattiSG) [inline test/spec/controllers/resultat.js]: > nearest

**Comment by MattiSG** [comment]
Clean job, GTM!

**Pull request merged**